### ISSUE TITLE
Fails for version comparisons <10 versus v10+

### ISSUE THREAD

**Issue by glebec** [typicode/please-upgrade-node#11]
[This line](https://github.com/typicode/please-upgrade-node/blob/8bbf4c8dee5d7af20685127977deb6cf8b92342f/index.js#L4):

```js
var requiredVersion = pkg.engines.node.replace(">=", "");
var currentVersion = process.version.replace("v", "");
if (currentVersion < requiredVersion) { // oops!
  ...
}
```

…is an error.

```js
'6' < '10.0.0' === false
```

You cannot use string comparison here, which sorts lexicographically. You'll need a bit more logic to compare version number strings.

**Comment by glebec** [comment]
One possible choice to fix this: https://github.com/substack/semver-compare. Haven't vetted it or anything, just figured others have had to compare semver strings in the past.

---

PS — see that issue #9 also gives a good suggestion as to a package to improve this.

**Comment by j-f1** [comment]
#10 should fix this.

**Comment by typicode** [comment]
Thanks for the heads up and the link to semver-compare @glebec, a fix is coming.
I didn't know about semver-compare, I may end up using it instead of semver as it's lighter.

**Comment by typicode** [comment]
Should be good in `v3.0.2`, thanks again @glebec and thanks also @ronkorving, @j-f1 for the PR/suggesting using something more robust 👍

**Comment by prichodko** [comment]
@typicode as long as it's a valid semver, this is enough for comparison:

```js
function compare(a, b) {
  return a > b ? 1 : a === b ? 0 : -1
}
```

**Comment by glebec** [comment]
@prichodko what is the context of your post? If `a` and `b` are supposed to be semver strings, that function does not properly order them. Strings sort lexicographically, not numerically or by server range. So `compare('6', '10')` will return `1` when it should return `-1`. Anyway, Typicode already solved this in 3.0.2.

**Comment by jesstelford** [comment]
@prichodko semver is represented as strings, not integers. When you do a `>` comparison, JS will do an alphabetic (aka _lexographic_) comparison letter by letter until it has a definitive answer.

So, with `"6" > "10"`, it first compares `"6" > "1"` alphabetically. Because `1` comes before `6` _alphabetically_, the comparison of `"6" > "10"` will complete and return `true`.
